### PR TITLE
MCLOUD-5526: Improve the detection of master branch in ECE-Tools

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-5526

### Manual testing scenarios

1. https://jira.corp.magento.com/browse/MAGECLOUD-18

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

releasenotes
Improved master branch detection mechanism. From now all branches beginning with `master`, `staging`, `production` will be treated as master branches

### CODE REVIEW

**Review by arhiopterecs — DISMISSED** [review]

File: src/Config/Environment.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ class Environment
      *   master - is for integration environment;
      *   production and staging are for production and staging environments respectively.
      */
-    private const GIT_MASTER_BRANCH_RE = '/^(master|production|staging)(?:-[a-z0-9]+)?$/i';
+    private const GIT_MASTER_BRANCH_RE = '/^(master|production|staging)(?:[a-z0-9-]+)?$/i';
```
Comment (oshmyheliuk) [inline src/Config/Environment.php]: Looks like we can left only `'/^(master|production|staging)/i'` as `(?:[a-z0-9-]+)?` contains all possible symbols in the branch name.

Comment (shiftedreality) [reply src/Config/Environment.php]: Agree, fixed

**Review by oshmyheliuk — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by NadiyaS** [comment]
@shiftedreality Devdocs should be updated too https://devdocs.magento.com/cloud/env/variables-deploy.html#force_update_urls

**Comment by andriyShevtsov** [comment]
QA approved

**Comment by meker12** [comment]
> @shiftedreality Devdocs should be updated too https://devdocs.magento.com/cloud/env/variables-deploy.html#force_update_urls

I didn't see any documentation changes submitted for this.

**Pull request merged**